web/react/components/get_link_modal.jsx

```diff
@@ -5,66 +5,120 @@ var UserStore = require('../stores/user_store.jsx');
 var ZeroClipboardMixin = require('react-zeroclipboard-mixin');
 
 ZeroClipboardMixin.ZeroClipboard.config({
-  swfPath: '../../static/flash/ZeroClipboard.swf'
+    swfPath: '../../static/flash/ZeroClipboard.swf'
 });
 
-module.exports = React.createClass({
-    displayName: 'GetLinkModal',
-    zeroclipboardElementsSelector: '[data-copy-btn]',
-    mixins: [ZeroClipboardMixin],
-    componentDidMount: function() {
+export default class GetLinkModal extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.zeroclipboardElementsSelector = '[data-copy-btn]';
+        this.mixins = [ZeroClipboardMixin];
+
+        this.handleClick = this.handleClick.bind(this);
+
+        this.state = {copiedLink: false};
+    }
+    componentDidMount() {
         var self = this;
```

Comment:
There's some more uses of self and getDOMNode in this function and below. You might want to grep through your files for these.